zerver/tests/test_audit_log.py

```diff
@@ -170,6 +170,14 @@ def test_change_password(self) -> None:
             ).count(),
             1,
         )
+
+        # changing password also changes the API key.
+        self.assertEqual(
+            RealmAuditLog.objects.filter(
+                event_type=RealmAuditLog.USER_API_KEY_CHANGED, event_time__gte=now
+            ).count(),
+            1,
+        )
```

Comment:
I think it makes more sense to directly test for the main effect rather than a "side-effect" - so to just verify that `user_profile.api_key` is different than before. Yeah, we create a `RealmAuditLog` entry in the process of changing the key, but that's not the main point we want to verify here.